File: code/appd-13.erl
-record(documentrecord, {docid, name}).

R = #documentrecord{docid=123, name="name"}.
  % -> #documentrecord{docid = 123,name = "name"}

R#documentrecord.name. % -> "name"

F = fun(#documentrecord{docid=DocId} = D) ->
            D#documentrecord{docid=DocId+1} end.
F(R). % -> #documentrecord{docid = 124,name = "name"}
R. % R is not changed -> #documentrecord{docid = 123,name = "name"}

File: code/appd-13.ex
require Record
Record.defrecord :documentrecord, docid: 0, name: ""
@type documentrecord::record(:documentrecord, docid: integer, name: String.t)

r = documentrecord(docid: 123,name: "name")
    # -> {:documentrecord, 123, "name"}

documentrecord(r,:name) # -> "name"

f = fn (d) ->
 documentrecord(docid: documentrecord(d,:docid)+1,name: documentrecord(d,:name))
end
f.(r) # -> {:documentrecord, 124, "name"}
r # r is not changed -> {:documentrecord, 123, "name"}
